node/src/ledger_effectful/ledger_effectful_effects.rs

```diff
@@ -0,0 +1,60 @@
+use redux::ActionWithMeta;
+
+use crate::{
+    ledger::{
+        read::{LedgerReadAction, LedgerReadInitCallback},
+        write::LedgerWriteAction,
+        LedgerService,
+    },
+    transition_frontier::sync::{
+        ledger::staged::TransitionFrontierSyncLedgerStagedAction, TransitionFrontierSyncAction,
+    },
+    BlockProducerAction, Store,
+};
+
+use super::LedgerEffectfulAction;
+
+pub fn ledger_effectful_effects<S>(
+    store: &mut Store<S>,
+    action: ActionWithMeta<LedgerEffectfulAction>,
+) where
+    S: LedgerService,
+{
+    let (action, _meta) = action.split();
+
+    match action {
+        LedgerEffectfulAction::WriteInit { request, on_init } => {
+            store.service.write_init(request.clone());
+            store.dispatch(LedgerWriteAction::Pending);
+            store.dispatch_callback(on_init, request);
+        }
+        LedgerEffectfulAction::WriteSuccess => {
```

Comment:
this one is not really an effect right? never calls any service, it just dispatches further actions

Reply:
btw the `ifs` here are just an optimization to avoid calling the enabling conditions for the other actions. In theory dispatching all the actions should be the same (assuming there are no bugs in the enabling conditions)